tests/assets/test_digital_assets.py

```diff
@@ -0,0 +1,192 @@
+
+
+@pytest.mark.usefixtures('inputs')
+def test_asset_transfer(b, user_vk, user_sk):
+    tx_input = b.get_owned_ids(user_vk).pop()
+    tx_create = b.get_transaction(tx_input['txid'])
+    tx_transfer = b.create_transaction(user_vk, user_vk, tx_input, 'TRANSFER')
+    tx_transfer_signed = b.sign_transaction(tx_transfer, user_sk)
+
+    assert b.validate_transaction(tx_transfer_signed) == tx_transfer_signed
+    assert tx_transfer_signed['transaction']['asset']['id'] == tx_create['transaction']['asset']['id']
+
+
+def test_validate_bad_asset_creation(b, user_vk):
+    from bigchaindb.util import get_hash_data
+    from bigchaindb.exceptions import AmountError
+
+    tx = b.create_transaction(b.me, user_vk, None, 'CREATE')
+    tx['transaction']['asset'].update({'divisible': 1})
+    tx['id'] = get_hash_data(tx['transaction'])
+    tx_signed = b.sign_transaction(tx, b.me_private)
+    with pytest.raises(TypeError):
+        b.validate_transaction(tx_signed)
+
+    tx = b.create_transaction(b.me, user_vk, None, 'CREATE')
+    tx['transaction']['asset'].update({'refillable': 1})
+    tx['id'] = get_hash_data(tx['transaction'])
+    tx_signed = b.sign_transaction(tx, b.me_private)
+    with pytest.raises(TypeError):
+        b.validate_transaction(tx_signed)
+
+    tx = b.create_transaction(b.me, user_vk, None, 'CREATE')
+    tx['transaction']['asset'].update({'updatable': 1})
+    tx['id'] = get_hash_data(tx['transaction'])
+    tx_signed = b.sign_transaction(tx, b.me_private)
+    with pytest.raises(TypeError):
+        b.validate_transaction(tx_signed)
+
+    tx = b.create_transaction(b.me, user_vk, None, 'CREATE')
+    tx['transaction']['asset'].update({'data': 'a'})
+    tx['id'] = get_hash_data(tx['transaction'])
+    tx_signed = b.sign_transaction(tx, b.me_private)
+    with pytest.raises(TypeError):
+        b.validate_transaction(tx_signed)
+
+    tx = b.create_transaction(b.me, user_vk, None, 'CREATE')
+    tx['transaction']['conditions'][0]['amount'] = 'a'
+    tx['id'] = get_hash_data(tx['transaction'])
+    tx_signed = b.sign_transaction(tx, b.me_private)
+    with pytest.raises(TypeError):
+        b.validate_transaction(tx_signed)
+
+    tx = b.create_transaction(b.me, user_vk, None, 'CREATE')
+    tx['transaction']['conditions'][0]['amount'] = 2
+    tx['transaction']['asset'].update({'divisible': False})
+    tx['id'] = get_hash_data(tx['transaction'])
+    tx_signed = b.sign_transaction(tx, b.me_private)
+    with pytest.raises(AmountError):
+        b.validate_transaction(tx_signed)
+
+    tx = b.create_transaction(b.me, user_vk, None, 'CREATE')
+    tx['transaction']['conditions'][0]['amount'] = 0
+    tx['id'] = get_hash_data(tx['transaction'])
+    tx_signed = b.sign_transaction(tx, b.me_private)
+    with pytest.raises(AmountError):
+        b.validate_transaction(tx_signed)
+
+
+@pytest.mark.usefixtures('inputs')
+def test_validate_bad_asset_transfer(b, user_vk, user_sk):
```

Comment:
A more descriptive test name might be nice... I think the validation is failing because of a bad asset id?